lib/compiler/src/beam_peep.erl

```diff
@@ -89,12 +89,34 @@ peep([{gc_bif,_,_,_,_,Dst}=I|Is], SeenTests0, Acc) ->
 peep([{jump,{f,L}},{label,L}=I|Is], _, Acc) ->
     %% Sometimes beam_jump has missed this optimization.
     peep(Is, gb_sets:empty(), [I|Acc]);
-peep([{select,Op,R,F,Vls0}|Is], _, Acc) ->
+peep([{select,Op,R,F,Vls0}|Is], SeenTests0, Acc) ->
     case prune_redundant_values(Vls0, F) of
 	[] ->
 	    %% No values left. Must convert to plain jump.
 	    I = {jump,F},
-	    peep(Is, gb_sets:empty(), [I|Acc]);
+	    peep([I|Is], gb_sets:empty(), Acc);
+        [{atom,_}=Value,Lbl] when Op =:= select_val ->
+            %% Single value left. Convert to regular test and pop redundant tests.
+            Is1 = [{test,is_eq_exact,F,[R,Value]},{jump,Lbl}|Is],
+            case Acc of
+                [{test,is_atom,F,[R]}|Acc1] ->
+                    peep(Is1, SeenTests0, Acc1);
+                _ ->
+                    peep(Is1, SeenTests0, Acc)
```

Comment:
This line is not covered.

Please try to construct a test case that will cover this line. I think that there will be always be a `test` instruction before the `select_val` instruction, but you should be able to construct a test case where the label for the `test` instruction is different from the label for the `select_val`.

(I suggest placing the test cases in either the existing `match_SUITE` module or in a new test module called `beam_peep_SUITE`.)